AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift

```diff
@@ -11,43 +11,204 @@
     }
-}
 
-// MARK: - Protocol Methods
-extension FullScreenMessage: FullScreenMessageUiInterface {
+    /// Creates `FullscreenMessage` instance with the payload provided.
+    /// WARNING: This API consumes HTML/CSS/JS using an embedded browser control.
+    /// This means it is subject to all the risks of rendering untrusted web pages and running untrusted JS.
+    /// Treat all calls to this API with caution and make sure input is vetted for safety somewhere.
+    ///
+    /// - Parameters:
+    ///     - payload: String html content to be displayed with the message
+    ///     - listener: `FullscreenMessaging` listener to listening the message lifecycle.
+    ///     - isLocalImageUsed: If true, an image from the app bundle will be used for the fullscreen message.
+    public init(payload: String, listener: FullscreenMessaging?, isLocalImageUsed: Bool) {
+        self.payload = payload
+        self.listener = listener
+        self.isLocalImageUsed = isLocalImageUsed
+    }
 
     public func show() {
-        fullScreenMessageHandler?.show()
+        if ServiceProvider.shared.messageMonitor.show() ==  false {
+            return
+        }
+
+        DispatchQueue.main.async {
+
+            guard var newFrame: CGRect = self.calcFullscreenFrame() else { return }
+            newFrame.origin.y = newFrame.size.height
+            if newFrame.size.width > 0.0 && newFrame.size.height > 0.0 {
+
+                let wkWebView = self.getConfiguredWebview(newFrame: newFrame)
+
+                // Fix for iPhone X to display content edge-to-edge
+                if #available(iOS 11, *) {
+                    wkWebView.scrollView.contentInsetAdjustmentBehavior = .never
+                }
+
+                // save the HTML payload to a local file if the cached image is being used
+                var useTempHTML = false
+                var cacheFolderURL: URL?
+                var tempHTMLFile: URL?
+                let cacheFolder: URL? = self.fileManager.getCacheDirectoryPath()
+                if cacheFolder != nil {
+                    cacheFolderURL = cacheFolder?.appendingPathComponent(self.DOWNLOAD_CACHE)
+                    tempHTMLFile = cacheFolderURL?.appendingPathComponent(self.TEMP_FILE_NAME).appendingPathExtension(self.HTML_EXTENSION)
+                    if !self.isLocalImageUsed {
+                        /* AMSDK-8942: The ACS extension saves downloaded remote image files in the cache. We have to use loadFileURL so we can allow read access to these image files in the cache but loadFileURL expects a file URL and not the string representation of the HTML payload. As a workaround, we can write the payload string to a temporary HTML file located at cachePath/adbdownloadcache/temp.html and pass that file URL to loadFileURL.
+                         */
+                        do {
+                            try FileManager.default.createDirectory(atPath: cacheFolderURL!.path, withIntermediateDirectories: true, attributes: nil)
```

Comment:
Even thoes these two lines are in a do/catch, It'd be preferable to remove the usage of the force unwrapping. `cacheFolderURL!.path` and `tempHTMLFile!.path`.